packages/cozy-stack-client/src/mangoIndex.js

```diff
@@ -46,26 +46,61 @@ export const normalizeDesignDoc = designDoc => {
   return { id, _id: id, ...designDoc.doc }
 }
 
+/**
+ * Process a partial filter to generate a string key
+ *
+ * @param {object} condition - An object representing the partial filter or a sub-condition of the partial filter
+ * @returns {string} - The string key of the processed partial filter
+ */
+export const processPartialFilter = condition => {
+  if (typeof condition !== 'object' || condition === null) {
+    return String(condition)
+  }
+
+  const conditions = Object.entries(condition).map(([key, value]) => {
+    if (key === '$or' || key === '$and') {
+      if (value.every(subObj => typeof subObj === 'string')) {
+        return `(${value.join(`_${key}_`)})`
+      } else {
+        return `(${value
+          .map(subCondition => `${processPartialFilter(subCondition)}`)
+          .join(`)_${key}_(`)})`
+      }
+    } else if (Array.isArray(value)) {
+      return `${key}_(${value.join('_')})`
+    } else if (typeof value === 'object') {
+      return `${key}_${processPartialFilter(value)}`
+    } else {
+      return `${key}_${value}`
+    }
+  })
+
+  return conditions.join(')_and_(')
+}
+
 /**
  * Name an index, based on its indexed fields and partial filter.
  *
  * It follows this naming convention:
- * `by_{indexed_field1}_and_{indexed_field2}_filter_{partial_filter_field1}_and_{partial_filter_field2}
+ * `by_{indexed_field1}_and_{indexed_field2}_filter_({partial_filter.key1}_{partial_filter.value1})_and_({partial_filter.key2}_{partial_filter.value2})`
  *
  * @param {Array<string>} fields - The indexed fields
- * @param {object} params - The additional params
- * @param {Array<string>=} params.partialFilterFields - The partial filter fields
+ * @param {object} [options] - The additional params
+ * @param {object} [options.partialFilter] - The partial filter
  * @returns {string} The index name, built from the fields
  */
-export const getIndexNameFromFields = (
-  fields,
-  { partialFilterFields } = {}
-) => {
+export const getIndexNameFromFields = (fields, options = {}) => {
```

Comment:
Why the generic and less explicit `options`  while only `partialFilter` is expected?

Reply:
It was a remnant of the previous function but you're right, it's clearer without the option.